manifests/resource/upstream/member.pp

```diff
@@ -37,26 +37,11 @@
 define nginx::resource::upstream::member (
   $upstream,
   $server,
-  $ensure                 = 'present',
-  $port                   = 80,
+  Enum['present', 'absent'] $ensure = 'present',
+  Integer $port                     = 80,
   $upstream_fail_timeout  = '10s',
 ) {
 
-  validate_re($ensure, '^(present|absent)$',
-    "${ensure} is not supported for ensure. Allowed values are 'present' and 'absent'.")
-
-  if is_string($port) {
-    warning('DEPRECATION: String $port must be converted to an integer. Integer string support will be removed in a future release.')
-  }
-  elsif !is_integer($port) {
-    fail('$port must be an integer.')
-  }
-
-  $ensure_real = $ensure ? {
```

Comment:
Not sure $ensure is used at all at the moment.  Use of `$ensure_real` was removed in https://github.com/voxpupuli/puppet-nginx/commit/e40ce7cea63f03c1a4c0076a773739a773f866cb ??